python_modules/dagster/dagster/_core/definitions/repository_definition.py

```diff
@@ -1295,6 +1297,57 @@ def source_assets_by_key(self) -> Dict[AssetKey, SourceAsset]:
     def _assets_defs_by_key(self) -> Mapping[AssetKey, "AssetsDefinition"]:
         return self._repository_data.get_assets_defs_by_key()
 
+    @public
+    def load_asset_value(
+        self,
+        asset_key: CoercibleToAssetKey,
+        python_type: Optional[Type] = None,
+        instance: Optional[DagsterInstance] = None,
+    ) -> object:
```

Comment:
should there be some way to apply resource config here? (not necessary for v0, but seems likely to be useful)

would also be nice to be able to specify partition key